antidotum/tergo/R/styling.R

```diff
@@ -21,83 +21,111 @@ style <- function(config_file = "tergo.toml", configuration = list(), ...) {
 #'
 #' @inheritParams style
-#' @param path (`character`) The path to the package.
-#' @param force (`logical(1`) Whether to format the files even
+#' @param path (`character`) The path to the package. Default `"."`.
+#' @param force (`logical(1)`) Whether to format the files even
 #' if no package was found. `TRUE` - format the `.R` and `.r` files
 #' found in the directory (recursive). `FALSE` exit without formatting
-#' anything.
-#' @param extensions (`character`) The extensions of the files to format.
+#' anything. Default `FALSE`.
+#' @param extensions (`character`) The extensions of the files to format. Default `c(".R", ".r")`.
+#' @param verbose (`logical(1)`) Whether per file status and run statistics should be printed. Default `interactive()`.
 #'
 #' @export
 style_pkg <- function(path = ".",
                       config_file = "tergo.toml",
                       configuration = list(),
                       force = FALSE,
-                      extensions = c(".R", ".r")) {
+                      extensions = c(".R", ".r"),
+                      verbose = interactive()) {
   if (!is.character(path) || length(path) != 1) {
-    stop("Path must be a character")
+    stop("Path must be a single character string.")
   }
   if (!is.character(config_file) || length(config_file) != 1) {
-    stop("Config file must be a character")
+    stop("Config file must be a single character string.")
   }
   if (!is.logical(force) || length(force) != 1) {
-    stop("Force must be a logical")
+    stop("Force must be a single logical value.")
   }
   if (!is.list(configuration)) {
-    stop("Configuration must be a list")
+    stop("Configuration must be a list.")
+  }
+  if (!is.logical(verbose) || length(verbose) != 1) {
+    stop("verbose must be a single logical value.")
   }
 
-  # Read a configuration file
+  # Read Configuration File
   wd <- path
   config <- NULL
 
   repeat {
-    if (file.exists(file.path(wd, config_file))) {
-      config <- file.path(wd, config_file)
+    config_path <- file.path(wd, config_file)
+    if (file.exists(config_path)) {
+      config <- config_path
       break
     }
-    # OS agnostic file path splitting
-    if (strsplit(wd, split = .Platform$file.sep) |> length() == 1) {
+    # Stop if at the root directory
+    if (dirname(wd) == wd) {
       break
     }
     wd <- dirname(wd)
   }
+
   if (!is.null(config)) {
     config <- get_config(config)
   } else {
     config <- get_default_config()
   }
   config[names(configuration)] <- configuration
 
-  # Find package root. If not found, then just list files in the current directory
+  # Find Package Root
   package_root <- getwd()
   while (!file.exists(file.path(package_root, "DESCRIPTION"))) {
-    package_root <- dirname(package_root)
-    if (strsplit(package_root, split = .Platform$file.sep) |> length() == 1) {
+    parent_dir <- dirname(package_root)
+    if (parent_dir == package_root) { # Reached root without finding DESCRIPTION
```

Comment:
Will this comparison behave nicely on Windows? I wouldn't want people to need to create two different styling scripts for Windows and Unix platforms because of backwards slashes.